rpmlint/checks/ProductCheck.py

```diff
@@ -0,0 +1,54 @@
+import re
+import stat
+import subprocess
+
+from xml.dom.minidom import parse
+from urllib.parse import unquote
+
+from rpmlint.checks.AbstractCheck import AbstractFilesCheck
+
+
+class ProductCheck(AbstractFilesCheck):
+    """
+    Validate that product files are correct. currently only cpeid.
+    """
+
+    def __init__(self, config, output):
+        super().__init__(config, output, r'/etc/products.d/.*\.prod$')
+
+    def check_file(self, pkg, filename):
+        cpeid_provider_found = None
+        cpeid_xml_found = None
+        for provide in pkg.provides:
+            if provide.name == 'product-cpeid()' and len(provide.version) > 1:
+                  cpeid_provider_found = unquote(provide.version[1])
+
+        if not cpeid_provider_found:
+            self.output.add_info('E', pkg, 'product-parsing-exception', 'no product-cpeid() provider', filename)
```

Comment:
this shouldn't be "product-parsing-exception". each error should have its own unique identifier. the format is "check-name", "additional information".

Reply:
done